### PR TITLE
chore(cypress): upgrade cypress to latest v12.5.1

### PR DESCRIPTION
**Pull request by aditya-radhakrishnan**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: smoke-test/tests/cypress/cypress/support/commands.js [hunk 1]
```diff
@@ -20,8 +20,10 @@ Cypress.Commands.add('login', () => {
       method: 'POST',
       url: '/logIn',
       body: {
-        username: Cypress.env('ADMIN_USERNAME'),
-        password: Cypress.env('ADMIN_PASSWORD'),
+        // username: Cypress.env('ADMIN_USERNAME'),
+        // password: Cypress.env('ADMIN_PASSWORD'),
+        username: 'datahub',
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress/support/commands.js:25]: please reset !

File: smoke-test/tests/cypress/package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "main": "index.js",
   "license": "MIT",
   "devDependencies": {
-    "cypress": "^9.7.0",
+    "cypress": "12.5.1",
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/package.json:7]: nice - thanks for pinning and locking this

File: smoke-test/tests/cypress/cypress.config.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const { defineConfig } = require('cypress')
+
+module.exports = defineConfig({
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress.config.js:3]: We might face a few challenges merging this back. Let's make sure that's accounted for !

File: smoke-test/tests/cypress/cypress/support/commands.js (hunk 1 above)
Comment (aditya-radhakrishnan) [reply smoke-test/tests/cypress/cypress/support/commands.js]: Done!

File: smoke-test/tests/cypress/integration_test.py [hunk 4]
```diff
@@ -165,7 +165,7 @@ def test_run_cypress(frontend_session, wait_for_healthchecks):
     else:
         record_arg = " "
 
-    rest_specs = set(os.listdir("tests/cypress/cypress/integration"))
+    rest_specs = set(os.listdir("tests/cypress/cypress/e2e"))
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/integration_test.py:168]: why are we changing this

**Review by jjoyce0510 — APPROVED** [review]

File: smoke-test/tests/cypress/integration_test.py (hunk 4 above)
Comment (aditya-radhakrishnan) [reply smoke-test/tests/cypress/integration_test.py:168]: Cypress just changed the directory they put these tests in, and they were automatically migrated

**Pull request merged**